From 0a01f8b35abc2ae558d5718670487949ec46f3d7 Mon Sep 17 00:00:00 2001
From: Alexander Martinz <amartinz@shiftphones.com>
Date: Wed, 4 Jul 2018 19:10:24 +0200
Subject: [PATCH 3/8] IconsHandler: prevent resource not found exception when
 getting xml

JavaDoc of Resources#getIdentifier(...):
  Returns - The associated resource identifier. Returns 0 if no such
            resource was found. (0 is not a valid resource ID.)

Change-Id: Ia9f4ea9efb56bb54d94729f70a3afcec5bce7999
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 src/com/android/launcher3/icons/IconsHandler.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/icons/IconsHandler.java b/src/com/android/launcher3/icons/IconsHandler.java
index 04c2dccc4..47a5b3aaf 100644
--- a/src/com/android/launcher3/icons/IconsHandler.java
+++ b/src/com/android/launcher3/icons/IconsHandler.java
@@ -215,7 +215,7 @@ public class IconsHandler {
             Resources res = mPackageManager.getResourcesForApplication(packageName);
             mCurrentIconPackRes = res;
             int resource = res.getIdentifier("drawable", "xml", packageName);
-            if (resource < 0) {
+            if (resource <= 0) {
                 return;
             }
             xpp = res.getXml(resource);
-- 
2.17.1

